app/assets/javascripts/admin/orders/controllers/bulk_cancel_controller.js.coffee

```diff
@@ -0,0 +1,14 @@
+angular.module("admin.orders").controller "bulkCancelCtrl", ($scope, $http, $timeout) ->
+
+  $scope.cancelOrder = (orderNum, sendEmailCancellation, restock_items) ->
+    $http(
+      method: 'GET'
+      url: "/admin/orders/#{orderNum}/fire?e=cancel&send_cancellation_email=#{sendEmailCancellation}&restock_items=#{restock_items}").then(->
+        window.location.reload()
+      )
+
+  $scope.cancelSelectedOrders = ->
+    ofnCancelOrderAlert((confirm, sendEmailCancellation, restock_items) ->
+      if confirm
+        $scope.cancelOrder orderNum, sendEmailCancellation for orderNum in $scope.selected_orders
```

Comment:
As I understand, we sent as requests there are orders to cancel. What about sending only one request with all the selected orders ids?

Reply:
I'm trying to create a new route called `admin/bulk_cancel` but I'm getting an unauthorized error. I don't understand why :shrug: 
```ruby
# config/routes/spree.rb
...
  match '/admin/bulk_cancel' => 'admin/orders#bulk_cancel', :as => "admin_bulk_cancel", via: :post
...
```
I get an `unauthorized` response when trying to make a request to the route

![unautho](https://user-images.githubusercontent.com/87677429/190880133-c8183550-62f7-4943-b019-dd7b756868f2.png)

Reply:
You mean
`match '/admin/orders/bulk_cancel' => 'admin/orders#bulk_cancel', :as => "admin_bulk_cancel", via: :post`

(missing *orders* in your path)

Reply:
I am still getting an `unauthorized` error. I pushed the code with the new route, could you take a look and tell me what I'm doing wrong here?

![unauthorized](https://user-images.githubusercontent.com/87677429/191147778-336f4d4f-fd37-48b1-9f89-aad444a8a77f.png)

Reply:
- Did you put a `debugger` in `app/controllers/spree/admin/base_controller.rb#authorize_admin`?

Reply:
Ok, maybe I've a clue, which might be a big bug, I should investigate.

Reply:
Found it, please see commit: 7d0a3680d1

Reply:
Thank You for your help JB. I should've been clearer and said that I was logged in as an enterprise user!

I've made the necessary changes as per your comments. Have a look and let me know if there's anything.